File: cPlusPlusCode/compareDifferentThresholdingOperationsImage.cpp
#include<opencv2/imgproc.hpp>
#include<opencv2/highgui.hpp>
#include<iostream>

using namespace std;
using namespace cv;

Mat image, result;
int thresholdValue=150;
int maxValue = 255;
int thresholdType = 1;
int maxType = 4;

void onTrackbarChangeValue(int , void*){
    threshold(image, result, thresholdValue, maxValue, thresholdType);
    imshow("image", result);
}

int main(){
    image = imread("../assets/putin.jpg", IMREAD_GRAYSCALE);

    namedWindow("image", WINDOW_NORMAL);

    createTrackbar("Type", "image", &thresholdType, maxType, onTrackbarChangeValue);
    createTrackbar("Value", "image", &thresholdValue, maxValue, onTrackbarChangeValue);

    onTrackbarChangeValue(0, 0);

    while(true){
        int k = waitKey(10);

        if(k == 27){
            break;
        }
    }
    
    destroyAllWindows();
    return 0;
}

File: pythonCode/compareDifferentThresholdingOperationsImage.py
import cv2

thresholdType = 1
thresholdValue = 150
maxType = 4
maxValue = 255

def onTrackbarTypeChange(*args):
    global thresholdType
    thresholdType = args[0]
    th, result = cv2.threshold(image, thresholdValue, maxValue, thresholdType)
    cv2.imshow("image", result)

def onTrackbarValueChange(*args):
    global thresholdValue
    thresholdValue = args[0]
    th, result = cv2.threshold(image, thresholdValue, maxValue, thresholdType)
    cv2.imshow("image", result)


image = cv2.imread("../assets/putin.jpg", cv2.IMREAD_GRAYSCALE)

cv2.namedWindow("image", cv2.WINDOW_NORMAL)

cv2.createTrackbar("Type", "image", thresholdType, maxType, onTrackbarTypeChange)

cv2.createTrackbar("Value", "image", thresholdValue, maxValue, onTrackbarValueChange)

onTrackbarTypeChange(1)
onTrackbarValueChange(150)

while(True):
    k=cv2.waitKey(10)
    if(k == 27):
        break


cv2.destroyAllWindows()